src/osd/PrimaryLogPG.cc

```diff
@@ -2385,17 +2385,28 @@ int PrimaryLogPG::do_manifest_flush(OpRequestRef op, ObjectContextRef obc, Flush
       bufferlist in;
       if (fp_oid != tgt_soid.oid) {
 	// decrement old chunk's reference count 
-	ObjectOperation dec_op;
-	cls_chunk_refcount_put_op put_call;
-	put_call.source = soid;
-	::encode(put_call, in);                             
-	dec_op.call("cas", "chunk_put", in);         
-	// we don't care dec_op's completion. scrub for dedup will fix this.
-	tid = osd->objecter->mutate(
-	  tgt_soid.oid, oloc, dec_op, snapc,
-	  ceph::real_clock::from_ceph_timespec(obc->obs.oi.mtime),
-	  flags, NULL);
-	in.clear();
+	SnapSetContext* ssc = obc->ssc;
+	bool has_reference = false;
+	if (ssc && ssc->snapset.clones.size() > 0) {
+	  if (check_has_ref_snap(soid, ssc->snapset.clone_overlap.rbegin()->first,
+		 iter->second)) {
+	    has_reference = true;
+	  }
+	} 
+	if (!has_reference) {
+	  ObjectOperation dec_op;
+	  cls_chunk_refcount_put_op put_call;
+	  put_call.source = soid;
```

Comment:
Can a single object have more than one reference to a chunk?  For instance, could two different 4k extents within the same object have the same fingerprint?

Reply:
Yes, for example, if object A contains chunk 1 and chunk 2, these two chunks can have the same contents. So, chunk 1's fingerprint will be AA and chunk 2's fingerprint will be AA.

Reply:
Ah, I suppose it doesn't matter because each of those references would have a refcount.